File: Medium/War/War.cs
using System;
using System.Collections;

class Solution
{
    static void Main(string[] args)
    {
        int n = int.Parse(Console.ReadLine()); 
        Queue player_one = new Queue(); 
        for (int i = 0; i < n; i++)
            player_one.Enqueue(Console.ReadLine());

        int m = int.Parse(Console.ReadLine());
        Queue player_two = new Queue(); 
        for (int i = 0; i < m; i++)
            player_two.Enqueue(Console.ReadLine());
        int counter = 0;
        Queue c1 = new Queue();
        Queue c2 = new Queue();
        

        while(true)
        {
            string a = player_one.Dequeue().ToString();
            string b = player_two.Dequeue().ToString();
            c1.Enqueue(a);
            c2.Enqueue(b);
            counter++;
            int round_winner = compare(a,b);

            if (round_winner ==3)
            {
                while (true)
                {
                    if (player_one.Count<4 || player_two.Count<4)
                    {
                        Console.WriteLine("PAT");
                        Environment.Exit(0);
                    }
                    for (int i = 0; i < 3; i++)
                        c1.Enqueue(player_one.Dequeue());
                    for (int i = 0; i < 3; i++)
                        c2.Enqueue(player_two.Dequeue());

                    a = player_one.Dequeue().ToString();
                    b = player_two.Dequeue().ToString();
                    c1.Enqueue(a);
                    c2.Enqueue(b);
                    round_winner = compare(a,b);
                    if (round_winner != 3)
                        break;
                }
            }
            int cc = c1.Count;
            if (round_winner == 1)
            {
                for (int i = 0; i < cc; i++)
                    player_one.Enqueue(c1.Dequeue());
                for (int i = 0; i < cc; i++)
                    player_one.Enqueue(c2.Dequeue());
            }
            else if (round_winner == 2)
            {
                for (int i = 0; i < cc; i++)
                    player_two.Enqueue(c1.Dequeue());
                for (int i = 0; i < cc; i++)
                    player_two.Enqueue(c2.Dequeue());
            }
            if (player_one.Count<1 || player_two.Count<1)
            {
                int h = player_two.Count<1 ? 1 : 2;
                Console.WriteLine($"{h} {counter}");
                break;
            }
        }
    }
    public static int convert(string a)
    {
        switch(a[0])
        {
            case '1': return 10;
            case 'J': return 11;
            case 'Q': return 12;
            case 'K': return 13;
            case 'A': return 14;
            default : return int.Parse(a[0].ToString());
        }
    }
    public static int compare(string a, string b)
    {
        int aa = convert(a);
        int bb = convert(b);
        if (aa>bb)
            return 1;
        else if (aa<bb)
            return 2;
        else
            return 3;
    }
}

File: Medium/War/War.py
import sys

def compare(a, b):
    aa, bb = 0, 0
    if   a[0] == '1': aa = int(10)
    elif a[0] == 'J': aa = int(11)
    elif a[0] == 'Q': aa = int(12)
    elif a[0] == 'K': aa = int(13)
    elif a[0] == 'A': aa = int(14)
    else:             aa = int(a[0])

    if   b[0] == '1': bb = int(10)
    elif b[0] == 'J': bb = int(11)
    elif b[0] == 'Q': bb = int(12)
    elif b[0] == 'K': bb = int(13)
    elif b[0] == 'A': bb = int(14)
    else:             bb = int(b[0])

    if   (aa>bb): return 1
    elif (aa<bb): return 2
    else:         return 3

n = int(input())  # the number of cards for player 1
player_one = []
for i in range(n):
    player_one.append(input())  # the n cards of player 1
m = int(input())  # the number of cards for player 2
player_two = []
for i in range(m):
    player_two.append(input())  # the m cards of player 2

counter = 0
c1 = []
c2 = []

while True:
    a = str(player_one.pop(0))
    b = str(player_two.pop(0))
    c1.append(a)
    c2.append(b)
    counter+=1
    round_winner = compare(a,b)

    if round_winner == 3:
        while True:
            if len(player_one)<4 or len(player_two)<4:
                print("PAT")
                exit()
            for i in range(3):
                c1.append(player_one.pop(0))
                c2.append(player_two.pop(0))

            a = str(player_one.pop(0))
            b = str(player_two.pop(0))
            c1.append(a)
            c2.append(b)
            round_winner = compare(a,b)
            if round_winner != 3:
                break

    cc = len(c1)
    if round_winner == 1:
        for i in range(cc):
            player_one.append(c1.pop(0))
        for i in range(cc):
            player_one.append(c2.pop(0))
    elif round_winner == 2:
        for i in range(cc):
            player_two.append(c1.pop(0))
        for i in range(cc):
            player_two.append(c2.pop(0))
    if len(player_one) < 1 or len(player_two) < 1:
        h = 1 if len(player_two) < 1 else 2
        print(h,counter)
        break